src/.editorconfig

```diff
@@ -0,0 +1,2 @@
+[*.{cs,vb}]
+dotnet_diagnostic.CA1305.severity = error		# Specify IFormatProvider
```

Comment:
Why enable this just in `src` and not repo-wide?

Reply:
Because of the suggestion here: https://github.com/xamarin/xamarin-android/pull/7993#issuecomment-1528451692

There were a lot of tests, build-tools, Mono.Options, etc. that would also need to be address or use `$(NoWarn)`.